src/profile/v100/app_preferences/security.rs

```diff
@@ -16,14 +16,18 @@ use crate::prelude::*;
 )]
 #[serde(rename_all = "camelCase")]
 #[display(
-    "cloud? {}, dev? {}",
+    "cloud? {}, dev? {}, advanced lock? {}",
     is_cloud_profile_sync_enabled,
-    is_developer_mode_enabled
+    is_developer_mode_enabled,
+    is_advanced_lock_enabled
 )]
 pub struct Security {
     pub is_cloud_profile_sync_enabled: bool,
```

Comment:
I would probably start using new types around bools now, easy to accidentally pass the wrong bool to the wrong value otherwise

Reply:
sorry, couldn't get what you meant here

Reply:
I mean we should change from:

```rust
pub struct Security {
   ...
    pub is_cloud_profile_sync_enabled: bool,
    pub is_developer_mode_enabled: bool,
    pub is_advanced_lock_enabled: bool,
...
}
```

To:

```rust
#[derive(..., )]
#[serde(transparent)]
pub struct IsCloudSyncEnabled(bool);

#[derive(..., )]
#[serde(transparent)]
pub struct IsDeveloperModeEnabled(bool);
pub struct IsAdvancedLockEnabled(bool);

uniffi::custom_newtype!(IsCloudSyncEnabled, bool);
uniffi::custom_newtype!(IsDeveloperModeEnabled, bool);
...

pub struct Security {
   ...
    pub is_cloud_profile_sync_enabled: IsCloudSyncEnabled,
    pub is_developer_mode_enabled: IsDeveloperModeEnabled,
    pub is_advanced_lock_enabled: IsAdvancedLockEnabled,
...
}
```

Making everything type safe :) 

(we seems suitable to use a macro for it perhaps! `decl newtype_bool!(IsDeveloperModeEnabled); `...